keel-core/src/test/java/com/netflix/rocket/api/artifact/internal/debian/DebianArtifactParserTests.java

```diff
@@ -0,0 +1,130 @@
+package com.netflix.rocket.api.artifact.internal.debian;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactStatus;
+import org.junit.jupiter.api.Test;
+
+public class DebianArtifactParserTests {
+  private DebianArtifactParser parser = new DebianArtifactParser();
+
+  @Test
+  void shouldParseStandardVersionString() {
+    String rawVersion =
+        "debian-local:pool/o/mypackage-server/mypackage-server_0.1.2-h2.afc245_all.deb";
+
+    assertThat(parser.parseStatus(rawVersion)).isEqualTo(ArtifactStatus.RELEASE);
+  }
+
+  @Test
+  void shouldParseLegacyVersionString() {
+    String rawVersion = "debian-local:pool/t/test-java-legacy_1.1-h1.9e74f98_all.deb";
+
+    assertThat(parser.parseStatus(rawVersion)).isEqualTo(ArtifactStatus.RELEASE);
+  }
+
+  @Test
+  void shouldReturnReleaseForLocalReleaseBuild() {
+    String rawVersion =
+        "debian-local:pool/o/mypackage-server/mypackage-server_1.100.5-LOCAL_all.deb";
+
+    assertThat(parser.parseStatus(rawVersion)).isEqualTo(ArtifactStatus.RELEASE);
+  }
+
+  @Test
+  void shouldParseCandidateVersionString() {
+    String rawVersion =
+        "debian-local:pool/o/mypackage-server/mypackage-server_0.1.2~rc.11-h2.afc245_all.deb";
+
+    assertThat(parser.parseStatus(rawVersion)).isEqualTo(ArtifactStatus.CANDIDATE);
+  }
+
+  @Test
+  void shouldReturnCandidateForLocalFinalBuild() {
+    String rawVersion =
+        "debian-local:pool/o/mypackage-server/mypackage-server_1.100.5~rc.23-LOCAL_all.deb";
+
+    assertThat(parser.parseStatus(rawVersion)).isEqualTo(ArtifactStatus.CANDIDATE);
+  }
+
+  @Test
+  void shouldParseSnapshotVersionString() {
+    String rawVersion =
+        "debian-local:pool/o/mypackage-server/mypackage-server_0.0.1176~dev.34267+2345af-h1886.4246bc3_all.deb";
```

Comment:
Does the string always have the `~` before `deb`? I noticed it's not in the regex like it is for `~rc`.

Reply:
I'm not sure, but I'm pretty sure this code is correct